index.php

```diff
@@ -43,6 +43,10 @@
 $coursecontext = context_course::instance($courseid);
 require_capability('block/opencast:viewunpublishedvideos', $coursecontext);
 
+// Invalidate Block cache;
+$cache = cache::make('block_opencast', 'videodata');
+$cache->delete($courseid);
```

Comment:
Rather invalidate by event instead of calling `delete`: https://docs.moodle.org/dev/Cache_API_-_Quick_reference#Invalidate_keys_using_an_event